thrift.js

```diff
@@ -123,17 +162,58 @@ Thrift.prototype._definitionProcessors = {
     Union: 'compileUnion'
 };
 
+Thrift.prototype.compileHeaders = function compileHeaders(defs) {
+    var self = this;
+    for (var index = 0; index < defs.length; index++) {
+        var def = defs[index];
+        var headerProcessor = self._headerProcessors[def.type];
+        // istanbul ignore else
+        if (headerProcessor) {
+            self[headerProcessor](def);
+        }
+    }
+};
+
 Thrift.prototype.compileDefinitions = function compileDefinitions(defs) {
     var self = this;
     for (var index = 0; index < defs.length; index++) {
         var def = defs[index];
+        var definitionProcessor = self._definitionProcessors[def.type];
         // istanbul ignore else
-        if (self._definitionProcessors[def.type]) {
-            self[self._definitionProcessors[def.type]](def);
+        if (definitionProcessor) {
+            self[definitionProcessor](def);
         }
     }
 };
 
+Thrift.prototype.compileInclude = function compileInclude(def) {
+    var self = this;
+
+    if (def.id.indexOf('./') === 0 || def.id.indexOf('../') === 0) {
+        var thriftFile = path.resolve(self.dirname, def.id);
+        var ns = def.namespace && def.namespace.name;
+
+        // If include isn't name, get filename sans *.thrift file extension.
+        if (!ns) {
+            var basename = path.basename(def.id);
+            ns = basename.slice(0, basename.length - 7);
+            if (!validThriftIdentifierRE.test(ns)) {
+                throw Error(
+                    'Thrift include filename is not valid thrift identifier'
+                );
+            }
+        }
```

Comment:
Okay. I'll add this after. I wanted to add this now so i could use filename like all the others in the test folder, since the current filenaming strategy doesn't product valid thrift identifiers. That was the impetus for including this now